File: easy/1704_Determine_if_String_Halves_Are_Alike/main.cpp
#include <string>
#include <algorithm>

using namespace std;

class Solution {
public:
    bool halvesAreAlike(string s) {
        int cnt = 0,n = s.size();
        transform(s.begin(), s.end(), s.begin(), ::tolower); 
        for(int  i = 0;i<n;i++)
        {
            if(s[i]=='a'||s[i]=='e'||s[i]=='i'||s[i]=='o'||s[i]=='u')
                cnt++;
        }
        int c = 0;
        for(int i = 0;i<n/2;i++)
        {
            if(s[i]=='a'||s[i]=='e'||s[i]=='i'||s[i]=='o'||s[i]=='u')
                c++;
        }
        if(2*c==cnt)
            return true;
        else
            return false;
    }
};

File: easy/1704_Determine_if_String_Halves_Are_Alike/main.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        vowels = set('aeiouAEIOU')
        nVA = nVB = 0
        i, j = 0, len(s) - 1
        while i < j:
            nVA += s[i] in vowels
            nVB += s[j] in vowels
            i += 1
            j -= 1
        return nVA == nVB
